### PR TITLE
chore: add readme template and readme generator ci steps

### PR DESCRIPTION
**Pull request by anitarua**
addresses #28 

adds readme template, contributing doc, and quickstart example to embed in the reamde. Also adds readme generator verification and generation steps in ci, but had to split those out into pull request vs push to main files.

### CODE REVIEW

File: .github/workflows/on-pull-request.yml [hunk 1]
```diff
@@ -6,8 +6,6 @@
 name: Dart
```
Comment (bruuuuuuuce) [inline .github/workflows/on-pull-request.yml:6]: Can we change this name to `Pull Request`

File: .github/workflows/on-push-to-main.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+
+name: Dart
```
Comment (bruuuuuuuce) [inline .github/workflows/on-push-to-main.yml:6]:
```suggestion
name: Push to Main
```

File: .github/workflows/on-push-to-main.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
+# This workflow uses actions that are not certified by GitHub.
```
Comment (bruuuuuuuce) [inline .github/workflows/on-push-to-main.yml:1]: just curious where this paragraph is coming from

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
+# Welcome to client-sdk-dart contributing guide :wave:
```
Comment (bruuuuuuuce) [inline CONTRIBUTING.md:1]: 👋

File: .github/workflows/on-push-to-main.yml (hunk 3 above)
Comment (anitarua) [reply .github/workflows/on-push-to-main.yml:1]: no idea lol, I'll remove that paragraph at the top of both

**Review by bruuuuuuuce — APPROVED** [review]
:shipit:

**Pull request merged**